client/layout/masterbar/logged-in.jsx

```diff
@@ -201,140 +223,272 @@ class MasterbarLoggedIn extends Component {
+		const { translate, user } = this.props;
+		return (
+			<Item
+				tipTarget="me"
+				url="/me"
+				icon={ isMobile ? null : 'user-circle' }
+				onClick={ this.clickMe }
+				isActive={ this.isActive( 'me' ) }
+				className="masterbar__item-me"
+				tooltip={ translate( 'Update your profile, personal settings, and more' ) }
+				preloadSection={ this.preloadMe }
+			>
+				<Gravatar
+					className="masterbar__item-me-gravatar"
+					user={ this.props.user }
+					alt={ translate( 'My Profile' ) }
+					size={ 18 }
 				/>
-			);
+				<span className="masterbar__item-me-label">
+					{ isMobile
+						? user?.display_name
+						: translate( 'My Profile', {
+								context: 'Toolbar, must be shorter than ~12 chars',
+						  } ) }
+				</span>
+			</Item>
+		);
+	};
+	renderNotifications = () => {
+		const { translate } = this.props;
+		return (
+			<Notifications
+				isShowing={ this.props.isNotificationsShowing }
+				isActive={ this.isActive( 'notifications' ) }
+				className="masterbar__item-notifications"
+				tooltip={ translate( 'Manage your notifications' ) }
+			>
+				<span className="masterbar__item-notifications-label">
+					{ translate( 'Notifications', {
+						comment: 'Toolbar, must be shorter than ~12 chars',
+					} ) }
+				</span>
+			</Notifications>
+		);
+	};
+	renderMenu = () => {
+		const { menuBtnRef } = this.state;
+		const { translate, hasDismissedThePopover, isFetchingPrefs } = this.props;
+		return (
+			<>
+				<Item
+					tipTarget="Menu"
+					icon="menu"
+					onClick={ this.handleToggleMenu }
+					isActive={ this.state.isMenuOpen }
+					className="masterbar__item-menu"
+					tooltip={ translate( 'Menu' ) }
+					ref={ ( ref ) => ref !== menuBtnRef && this.setState( { menuBtnRef: ref } ) }
+				/>
+				<MasterBarMobileMenu onClose={ this.handleToggleMenu } open={ this.state.isMenuOpen }>
+					{ this.renderPublish() }
+					{ this.renderReader() }
+					{ this.renderMe() }
+				</MasterBarMobileMenu>
+				{ menuBtnRef && (
+					<Popover
+						className="masterbar__new-menu-popover"
+						isVisible={ ! isFetchingPrefs && ! hasDismissedThePopover }
+						context={ menuBtnRef }
+						onClose={ this.dismissPopover }
+						position="bottom left"
+						showDelay={ 500 }
+					>
+						<div className="masterbar__new-menu-popover-inner">
+							<h1>
+								{ translate( '👆 New top navigation', {
+									comment: 'This is a popover title under the masterbar',
+								} ) }
+							</h1>
+							<p>{ translate( 'We changed the navigation for a cleaner experience.' ) }</p>
+							<div className="masterbar__new-menu-popover-actions">
+								<Button onClick={ this.dismissPopover }>
+									{ translate( 'Got it', { comment: 'Got it, as in OK' } ) }
+								</Button>
+							</div>
+						</div>
+					</Popover>
+				) }
+			</>
+		);
+	};
+	renderPopupSearch = () => {
+		const isWordPressActionSearchFeatureEnabled = config.isEnabled( 'wordpress-action-search' );
```

Comment:
Nit, but I've generally found it more readable to `return` early if we won't render anything, rather than using a ternary. For example:

```
if( ! config.isEnabled( 'wordpress-action-search' ) || ! isActionSearchVisible ) {
	return null;
}
```

followed by the `return <AsyncLoad ...` bit.